src/core/jit/context.rs

```diff
@@ -8,21 +10,50 @@ use crate::core::ir::ResolverContextLike;
 pub struct Context<'a, Input, Output> {
     request: &'a Request<Input>,
     value: Option<&'a Output>,
+    args: Option<&'a indexmap::IndexMap<&'a str, Input>>,
+    arg_const_value: Option<Arc<indexmap::IndexMap<async_graphql::Name, async_graphql::Value>>>,
 }
 
 impl<'a, Input, Output> Clone for Context<'a, Input, Output> {
     fn clone(&self) -> Self {
-        Self { request: self.request, value: self.value }
+        Self {
+            request: self.request,
+            value: self.value,
+            args: self.args,
+            arg_const_value: None,
+        }
     }
 }
 
 impl<'a, Input, Output> Context<'a, Input, Output> {
     pub fn new(request: &'a Request<Input>) -> Self {
-        Self { request, value: None }
+        Self { request, value: None, args: None, arg_const_value: None }
     }
 
     pub fn with_value(&self, value: &'a Output) -> Self {
-        Self { request: self.request, value: Some(value) }
+        Self {
+            request: self.request,
+            value: Some(value),
+            args: self.args,
+            arg_const_value: None,
+        }
+    }
+
+    pub fn with_args(&self, args: &'a indexmap::IndexMap<&str, Input>) -> Self
+    where
+        async_graphql::Value: From<Input>,
+        Input: Clone,
+    {
+        let mut map = indexmap::IndexMap::new();
+        for (key, value) in args.iter() {
+            map.insert(async_graphql::Name::new(key), value.clone().into());
+        }
+        Self {
+            request: self.request,
+            value: self.value,
+            args: Some(args),
+            arg_const_value: Some(Arc::new(map)),
```

Comment:
why store `args` and `arg_const_value` separately and why use only the second one later?